editor/src/clj/editor/code/view.clj

```diff
@@ -2641,6 +2684,129 @@
+    {:fx/type fxui/with-popup
+     :desc {:fx/type fxui/ext-value :value (.getWindow (.getScene canvas))}
+     :showing true
+     :anchor-x (.getX anchor)
+     :anchor-y (.getY anchor)
+     :anchor-location :window-top-left
+     :auto-hide true
+     :auto-fix true
+     :hide-on-escape true
+     :on-auto-hide {:event :cancel}
+     :consume-auto-hiding-events true
+     :event-handler consume-breakpoint-popup-events
+     :content
+     [{:fx/type fx.v-box/lifecycle
+       :stylesheets [(str (io/resource "editor.css"))]
+       :children
+       [{:fx/type fx.v-box/lifecycle
+         :style-class "breakpoint-editor"
+         :fill-width false
+         :spacing -1
+         :children
+         [{:fx/type fx.region/lifecycle
+           :view-order -1
+           :style-class "breakpoint-editor-arrow"
+           :min-width 10
+           :min-height 10}
+          {:fx/type fx.stack-pane/lifecycle
+           :children
+           [{:fx/type fx.region/lifecycle
+             :style-class "breakpoint-editor-background"}
+            {:fx/type fx.v-box/lifecycle
+             :style-class "breakpoint-editor-content"
+             :spacing padding
+             :children
+             [{:fx/type fx.label/lifecycle
+               :style-class ["label" "breakpoint-editor-label" "breakpoint-editor-header"]
+               :text (format "Breakpoint on line %d" (inc ^long (data/breakpoint-row edited-breakpoint)))}
+              {:fx/type fx.h-box/lifecycle
+               :spacing spacing
+               :alignment :baseline-left
+               :children
+               [{:fx/type fx.label/lifecycle
+                 :style-class ["label" "breakpoint-editor-label"]
+                 :text "Condition"}
+                {:fx/type fxui/text-field
+                 :h-box/hgrow :always
+                 :style-class ["text-field" "breakpoint-editor-label"]
+                 :prompt-text "e.g. i == 1"
+                 :text (:condition edited-breakpoint "")
+                 :on-text-changed {:event :edit}
+                 :on-action {:event :apply}}]}
+              {:fx/type fx.h-box/lifecycle
+               :spacing spacing
+               :alignment :center-right
+               :children
+               [{:fx/type fx.button/lifecycle
+                 :style-class ["button" "breakpoint-editor-button"]
+                 :text "OK"
+                 :on-action {:event :apply}}]}]}]}]}]}]}))
+
+(defn- create-breakpoint-editor! [view-node canvas ^Tab tab]
+  (let [state (atom nil)
+        timer (ui/->timer
+                10
+                "breakpoint-code-editor-timer"
+                (fn [_ _ _]
+                  (when (and (.isSelected tab) (not (ui/ui-disabled?)))
+                    (g/with-auto-evaluation-context evaluation-context
+                      (reset! state
+                        (if-let [edited-breakpoint (g/node-value view-node :edited-breakpoint evaluation-context)]
```

Comment:
We could use `when-some` instead of `if-let` here, since we're returning `nil` when false.

Reply:
Hmm yes, I will use `when-let` instead of `if-let`. 

Btw, I'm usually hesitant to use `when-some` because I assume it will do more work than `when-let`. `(when-let [x ...] ...)` expands to `(if x ...)`, which checks if x is not nil or false in `if`, but `(when-some [x ...] ...)` expands to `(if (nil? x) nil ...)`, which will first check if `x` is not nil in `nil?`, and then it will check if a boolean returned by that check is not nil or false in `if`. 

Because of that assumption, I default to `if-let` and `when-let`. Maybe I'm thinking too much into this?

Reply:
That's true. And while I don't think it would make much of a difference most of the time, it is something to keep in mind. I tend to prefer `if-some` and `when-some` because I think it is somewhat clearer, but maybe I should start using `if-let` and `when-let` unless the bound values is a boolean. 🤔